r/aws_msk_sasl_scram_secret: New Resource (#15302)
hashicorp/terraform-provider-aws: website/docs/r/msk_sasl_scram_secret.markdown

---
subcategory: "Managed Streaming for Kafka (MSK)"
layout: "aws"

> page_title: "AWS: aws_msk_sasl_scram_secret"

Reviewer: small typo given the resource is now `aws_msk_scram_secret`

Reviewer: also, let's update the file name as well to match the resource 👍

## Example Usage

```hcl
resource "aws_msk_cluster" "example" {
  cluster_name           = "example"
  kafka_version          = "2.4.1"
  number_of_broker_nodes = 3

  broker_node_group_info {
    instance_type   = "kafka.m5.large"
    ebs_volume_size = 1000
    client_subnets = [
      aws_subnet.subnet_az1.id,
      aws_subnet.subnet_az2.id,
      aws_subnet.subnet_az3.id,
    ]
    security_groups = [aws_security_group.sg.id]
  }
  client_authentication {
    sasl {
      scram = true
    }
  }
}

resource "aws_secretsmanager_secret" "example" {
  name       = "AmazonMSK_example"
  kms_key_id = aws_kms_key.example.key_id
}

resource "aws_secretsmanager_secret_version" "example" {
  secret_id     = aws_secretsmanager_secret.example.id
  secret_string = jsonencode({ username = "example_user", password = "example_password" })
}

resource "aws_kms_key" "example" {
  description = "example"
}

> resource "aws_msk_sasl_scram_secret" "msk" {

Reviewer: ```suggestion
resource "aws_msk_scram_secret" "msk" {
```

  cluster_arn     = aws_msk_cluster.example.arn
  secret_arn_list = [aws_secretsmanager_secret.example.arn]
}

resource "aws_secretsmanager_secret_policy" "msk" {
  secret_arn = aws_secretsmanager_secret.msk.arn
  policy     = <<POLICY
{
  "Version" : "2012-10-17",
  "Statement" : [ {
    "Sid": "AWSKafkaResourcePolicy",
    "Effect" : "Allow",
    "Principal" : {
      "Service" : "kafka.amazonaws.com"
    },
    "Action" : "secretsmanager:getSecretValue",
    "Resource" : "${aws_secretsmanager_secret.msk.arn}"
  } ]
}
POLICY
}
```

## Argument Reference

The following arguments are supported:

> * `cluster_arn` - (Required) Arn of the Kafka cluster.

Reviewer: Nit: 
```suggestion
* `cluster_arn` - (Required) Amazon Resource Name (ARN) of the Kafka cluster.
```

> * `secret_arn_list` - (Required) List of AWS Secret Manager secrets Arns to be associated.

Reviewer: ```suggestion
* `secret_arn_list` - (Required) List of AWS Secrets Manager secret Amazon Resource Names (ARNs) to be associated.
```

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

> * `scram_secrets` - Amazon Resource Name (ARN) of the MSK cluster.

Reviewer: ```suggestion
* `id` - Amazon Resource Name (ARN) of the MSK cluster.
```

Reviewer 2: Sorry for the interference, but I think `id` is missleading. ID and ARN are distinct terms in AWS' terminology. I would argue for using `arn` when this really means an ARN.

Reviewer: np @xenji! happy to hear your thoughts on the `arn` field as well since it's proposed in this resource's schema but i feel it is redundant as it too represents the MSK Cluster ARN
